backend/test/course_test.js

```diff
@@ -151,3 +152,94 @@ describe("Test assign course to curriculumn", () => {
     await conn.release();
   });
 });
+
+describe("Tests add student future desired courses", function() {
+
+  username = "MathieuTest";
+  password = "Mat123!@#";
+  email = "mat.test@mcgill.ca";
+  id = "192837465";
+
+  invalid_id_1 = "260111111"
+  invalid_id_2 = "26280x028"
+
+  before(async () => {
+
+    await users.insertStudentUser (username, password, email, id);
+
+  });
+
+  it("responds with invalid format student id 1", function(done) {
+    courses
+      .saveUserPreferences(null, ["ECSE 307", "ECSE 251", "ECSE 325"])
+      .catch(response => {
+        return new Promise(function(resolve) {
+          assert.equal(response.message, "Id cannot be empty");
```

Comment:
Nice job matt,quick question why do you return a promise?

Reply:
followed legacy code implementation, I just made changes to them